src/seedu/addressbook/data/person/Address.java

```diff
@@ -25,10 +33,20 @@ public Address(String address, boolean isPrivate) throws IllegalValueException {
         if (!isValidAddress(address)) {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
-        this.value = address;
+        String[] addressParts = splitAdress(address);
+        System.out.println(addressParts.length);
+        this.block = new Block(addressParts[ADDRESS_BLOCK_INDEX]);
+        this.street = new Street(addressParts[ADDRESS_UNIT_INDEX]);
+        this.unit = new Unit(addressParts[ADDRESS_STREET_INDEX]);
+        this.postalCode = new PostalCode(addressParts[ADDRESS_POSTAL_CODE_INDEX]);
     }
 
-    /**
+    private String[] splitAdress(String address) {
+		String[] addressParts = address.split(",");
```

Comment:
Indentation problem here?. Our coding standard requires you to use 4 spaces instead of tabs. You can configure Eclipse to convert tabs to spaces automatically.